cfn-resources/data-lake-pipeline/cmd/resource/resource.go

```diff
@@ -0,0 +1,353 @@
+		}
+	}
+
+	for i, transformation := range currentModel.Transformations {
+		dataLakeIntegrationPipeline.Transformations[i] = admin.FieldTransformation{
+			Field: transformation.Field,
+			Type:  transformation.Type,
+		}
+	}
+
+	return &dataLakeIntegrationPipeline
+}
+
+// Read handles the Read event from the Cloudformation service.
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(ReadRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlasClient, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	groupID := *currentModel.ProjectId
+	pipelineName := *currentModel.Name
+
+	readRequest := atlasClient.AtlasV2.DataLakePipelinesApi.GetPipeline(ctx.Background(), groupID, pipelineName)
+	pe, response, err := readRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		return handleError(response, err)
+	}
+
+	model := ReadResponseModelGeneration(pe)
+	model.Profile = currentModel.Profile
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read Completed",
+		ResourceModel:   model}, nil
+}
+
+func ReadResponseModelGeneration(pe *admin.DataLakeIngestionPipeline) (model *Model) {
+	if pe != nil {
+		source := Source{
+			Type:           pe.Source.Type,
+			ClusterName:    pe.Source.ClusterName,
+			CollectionName: pe.Source.CollectionName,
+			DatabaseName:   pe.Source.DatabaseName,
+			GroupId:        pe.Source.GroupId,
+		}
+
+		partitionArr := []PartitionFields{}
+
+		for i := range pe.Sink.PartitionFields {
+			partitionField := PartitionFields{
+				FieldName: &pe.Sink.PartitionFields[i].FieldName,
+				Order:     &pe.Sink.PartitionFields[i].Order,
+			}
+			partitionArr = append(partitionArr, partitionField)
+		}
+		sink := Sink{
+			Type:             pe.Sink.Type,
+			MetadataProvider: pe.Sink.MetadataProvider,
+			MetadataRegion:   pe.Sink.MetadataRegion,
+			PartitionFields:  partitionArr,
+		}
+
+		transformationsArr := []Transformations{}
+		for i := range pe.Transformations {
+			transformations := Transformations{
+				Field: pe.Transformations[i].Field,
+				Type:  pe.Transformations[i].Type,
+			}
+			transformationsArr = append(transformationsArr, transformations)
+		}
+		createdStr := pe.CreatedDate.Format(time.RFC3339)
+		lastUpdatedStr := pe.LastUpdatedDate.Format(time.RFC3339)
+
+		models := Model{
+			ProjectId:       pe.GroupId,
+			Name:            pe.Name,
+			Id:              pe.Id,
+			CreatedDate:     &createdStr,
+			LastUpdatedDate: &lastUpdatedStr,
+			Sink:            &sink,
+			Source:          &source,
+			Transformations: transformationsArr,
+		}
+		return &models
+	}
+	return &Model{}
```

Comment:
if pe *admin.DataLakeIngestionPipeline is nill we would be returning an empty model, leading to unexpected errors, i think this function should be like this func ReadResponseModelGeneration(pe admin.DataLakeIngestionPipeline) (model *Model)  and every-time you are calling the resource check if is null, if is null ill return an error

Reply:
Instead of empty model object return nil and validated nil value in its reference place